### PR TITLE
Capsule Layer (With Dynamic Routing Between Capsules)

### PR DESCRIPTION
**Pull request by Silver-Whiskey**
This is an implementation of capsule neural networks as described in "Dynamic Routing Between Capsules" paper (https://arxiv.org/abs/1710.09829).

I made a capsule layer, which was not originally implemented in Tensorflow, which made the developers and researchers job a bit tedious.

I have implemented fuctions given below:
1.) __init__ :  where you can do all input-independent initialization
2.) build: where you know the shapes of the input tensors and can do the rest of the initialization
3.) call: where you know the shapes of the input tensors and can do the rest of the initialization
4.) squash: an activation function discussed in the paper
5.) softmax: re-implemented softmax
6.) get_config: this function is important for storing the layer config while saving the model.
7.) compute_output_shape: This function helps in parsing the input dimension to the next layer

### CODE REVIEW

**Comment by gbaned** [comment]
It looks like your PR relates to the Keras component. Please submit it to the [github.com/keras-team/keras](github.com/keras-team/keras) repository instead. Thankyou.

**Comment by Silver-Whiskey** [comment]
> @Silver-Whiskey It looks like your PR relates to the Keras component. Please submit it to the [github.com/keras-team/keras](github.com/keras-team/keras) repository instead. Thankyou.

Ohhh thanks ! I thought all new contributions to keras should be made here. I have made the PR there: https://github.com/keras-team/keras/pull/15287
